### PR TITLE
Switch to NoArtifactContentViewSet from pulpcore

### PR DESCRIPTION
**Pull request by quba42**
[noissue]

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # All things django and asyncio are deliberately left to pulpcore
 # Example transitive requirements: asgiref, asyncio, aiohttp
-pulpcore>=3.45.1,<3.55
+pulpcore>=3.49.0,<3.55
```
Comment (quba42) [inline requirements.txt:3]: Can be merged once pulpcore 3.49.0 is released.

See https://pypi.org/project/pulpcore/

File: setup.py [hunk 2]
```diff
@@ -29,8 +29,10 @@
         "Framework :: Django",
         "Programming Language :: Python",
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
+        "Programming Language :: Python :: 3.12",
```
Comment (quba42) [inline setup.py:35]: Switched to the same list of supported python versions as pulpcore 3.49.0.

Not sure if this requires any kind of special testing or maybe a changelog now?

File: requirements.txt (hunk 1 above)
Comment (quba42) [reply requirements.txt:3]: Is available on pypi

File: setup.py (hunk 2 above)
Comment (quba42) [reply setup.py:35]: Added a changelog and more required changes.

**Review by pedro-psb — APPROVED** [review]
LGTM

**Review by ggainey — APPROVED** [review]

**Pull request merged**